apps/remote_control/lib/lexical/remote_control/search/indexer/extractors/module.ex

```diff
@@ -0,0 +1,166 @@
+  @moduledoc """
+  Extracts module references and definitions from AST
+  """
+
+  alias Lexical.Ast.Aliases
+  alias Lexical.ProcessCache
+  alias Lexical.RemoteControl.Search.Indexer.Entry
+  alias Lexical.RemoteControl.Search.Indexer.Metadata
+  alias Lexical.RemoteControl.Search.Indexer.Source.Block
+  alias Lexical.RemoteControl.Search.Indexer.Source.Reducer
+
+  # extract a module definition
+  def extract(
+        {:defmodule, defmodule_meta,
+         [{:__aliases__, module_name_meta, module_name}, module_block]},
+        %Reducer{} = reducer
+      ) do
+    %Block{} = block = Reducer.current_block(reducer)
+    aliased_module = resolve_alias(reducer, module_name)
+
+    entry =
+      Entry.definition(
+        reducer.document.path,
+        block.ref,
+        block.parent_ref,
+        aliased_module,
+        :module,
+        reducer.position,
+        block.ends_at,
+        Application.get_application(aliased_module),
+        &tokenize/1
+      )
+
+    module_name_meta = Reducer.skip(module_name_meta)
+
+    elem =
+      {:defmodule, defmodule_meta, [{:__aliases__, module_name_meta, module_name}, module_block]}
+
+    {:ok, entry, elem}
+  end
+
+  # This matches an elixir module reference
+  def extract({:__aliases__, metadata, maybe_module}, %Reducer{} = reducer)
+      when is_list(maybe_module) do
+    case module(reducer, maybe_module) do
+      {:ok, module} ->
+        start = Metadata.position(metadata)
+        finish = Metadata.position(metadata, :last)
+
+        %Block{} = current_block = Reducer.current_block(reducer)
+
+        entry =
+          Entry.reference(
+            reducer.document.path,
+            make_ref(),
+            current_block.ref,
+            module,
+            :module,
+            start,
+            finish,
+            Application.get_application(module),
+            &tokenize/1
+          )
+
+        {:ok, entry}
+
+      _ ->
+        :ignored
+    end
+  end
+
+  # This matches an erlang module, which is just an atom
+  def extract(atom_literal, %Reducer{} = reducer) when is_atom(atom_literal) do
+    case module(reducer, atom_literal) do
+      {:ok, module} ->
+        %Block{} = current_block = Reducer.current_block(reducer)
+        {start_line, start_column} = start = reducer.position
+        atom_length = module |> Atom.to_string() |> String.length()
+        finish = {start_line, start_column + atom_length}
+
+        entry =
+          Entry.reference(
+            reducer.document.path,
+            make_ref(),
+            current_block.ref,
+            module,
+            :module,
+            start,
+            finish,
+            Application.get_application(module),
+            &tokenize/1
+          )
+
+        {:ok, entry}
+
+      :error ->
+        :ignored
+    end
+  end
+
+  def extract(_, _) do
+    :ignored
+  end
+
+  defp resolve_alias(%Reducer{} = reducer, unresolved_alias) do
```

Comment:
While it's using `at_ast` now, this function is still extremely similar to [`Ast.expand_alises`](https://github.com/lexical-lsp/lexical/blob/b25b988c074de13279027bb646408fa218f0d19e/apps/common/lib/lexical/ast.ex#L221). From a contributor's perspective, it's a lot easier to understand existing code and implement new features when common APIs are used -- can we either modify `expand_aliases` to use `at_ast` or create a `expand_alises_at_ast` that can be used instead of this function?

Reply:
I saw that you did some work in this area in your other PR, so I kept this here until that lands. `Aliases.at` does things a little differently than `Ast` (it uses `Code.Fragment`, for example), and I'd like to unify the two.